pkg/proxy/proxy.go

```diff
@@ -199,8 +186,22 @@ func (p *Proxy) RoundTrip(req *http.Request) (*http.Response, error) {
 	// Set up impersonation request.
 	rt := transport.NewImpersonatingRoundTripper(*conf, p.clientTransport)
 
+	req, remoteAddr := context.RemoteAddr(req)
+	serverDuration := time.Now()
+	clientDuration := context.ClientRequestTimestamp(req)
+
 	// Push request through round trippers to the API server.
-	return rt.RoundTrip(req)
+	resp, err := rt.RoundTrip(req)
+
+	var statusCode int
+	if resp != nil {
+		statusCode = resp.StatusCode
+	}
```

Comment:
So we will report `statusCode` as nil if there was an error in round-tripping? What will the client see in that case?

Reply:
The client will see the error code that comes from the error handler if the there _is_ an error.

If the response is nil then we have to report the status code as 0.

I actually made a mistake here that if there is an error, then we will be observing a client call twice (once here, and again in the error handler). I have now wrapped the client observation in an `err != nil`. The server still need to be observed here regardless.

Reply:
Is there some wrapper we could make the observation for both the error case (with the real error code) and the success case? I guess we lose access to some of the labels if we do that?

Reply:
Yeah exactly, the error handler code path isn't in most requests so we can't invoke it there, and indeed we loose some info.